### ISSUE TITLE
Link - Provide alternative text for icons used in `icon-end` or `icon-start` - (2036569908)

### ISSUE THREAD

**Issue by dqateam** [Esri/calcite-design-system#5573]
### Module: 
 14b Link 

### Violation: 
 Provide alternative text for images 

![image](https://user-images.githubusercontent.com/5023024/202761895-06e3e041-3511-4b08-8465-de250fa6ea14.png)

### WCAG Reference: 
- [1.1.1 Non-text Content](https://www.w3.org/WAI/WCAG21/Understanding/non-text-content.html) 

### Instance ID: 
 2036569908 

### Severity: 
 10 

### Description: 
 [Issue]
The "opens in a new window"  meaningful image without a textual equivalent. 

[User Impact]
Screen reader users will be unable to determine what these images represent.

[Code Reference] 
 ```
<calcite-icon class="calcite-link--icon icon-end" aria-hidden="true" icon="launch" scale="s" calcite-hydrated="">
<svg class="svg" fill="currentColor" height="100%" viewBox="0 0 16 16" width="100%" xmlns="http://www.w3.org/2000/svg">
.......
</svg>
</calcite-icon>
 ``` 

### Note: 
 [Suggestion]
Provide alternative text for images. Meaningful images must have a concise but descriptive textual equivalent. In this case remove aria-hidden="true" from ``` <Calcite-icon> ``` element. In the case of svg elements, it is recommended to add role="img", ``` <title> ``` with the text equivalent and aria-labelledby set to the id of the title element

[Compliant Code Example] 
 ```
<calcite-icon class="calcite-link--icon icon-end" aria-hidden="true" icon="launch" scale="s" calcite-hydrated="">
<svg role="img" aria-labelledby="svg-title" class="svg" fill="currentColor" height="100%" viewBox="0 0 16 16" width="100%" xmlns="http://www.w3.org/2000/svg">
<title id="svg-title>opens in a new window</title>
.......
</svg>
</calcite-icon>
 ```
 

### Media Type: 
 Images 

### Additional Resources: 
- Report Source: Product accessibility evaluation conducted on the Calcite Design System in August, 2022.

**Comment by geospatialem** [comment]
While this issue pertains to adding an icon, we could also tackle adding context when the `target="_blank"` is used with `link` and `button` components. 

A [Codepen](https://codepen.io/geospatialem/pen/eYrzqer) for context that adds assistive technology helpers when `target="_blank"`:

> (opens in a new tab)

**Comment by geospatialem** [comment]
For next steps, we should consider either a `label` property for additional context -or- consider the `rel` with `"external"` value for additional context to AT.

**Comment by geospatialem** [comment]
Updating to be a doc update so users know how to implement this (re: https://github.com/Esri/calcite-design-system/issues/5573#issuecomment-1320301162), rather than as an enhancement as native HTML does not support this by default.

**Comment by rpanichakit9541** [comment]
@geospatialem - Can you clarify how we would satisfy this A11y requirement and add aria-label to the svg element inside calcite-icon when using `calcite-notice` like below?

```typescript
<calcite-notice open icon="information">
            <div slot="message">Some messages</div>
</calcite-notice>
```

**Comment by geospatialem** [comment]
> @geospatialem - Can you clarify how we would satisfy this A11y requirement and add aria-label to the svg element inside calcite-icon when using `calcite-notice` like below?
> 
> ```ts
> <calcite-notice open icon="information">
>             <div slot="message">Some messages</div>
> </calcite-notice>
> ```

@rpanichakit9541 Calcite hides the icon from assistive technologies so they can't be discerned as they are supplemental to the component, so the goal would be to provide context in the component's slots for users to discern the contents. For instance:

```html
<calcite-notice open kind="danger" icon>
    <div slot="title">There are unresolved errors in the form</div>
    <div slot="message">Our helpful operators are available by phone to help</div>
</calcite-notice>
```

As far as the above issue is concerned WRT to `calcite-link`, [here's an example](https://github.com/Esri/calcite-design-system/issues/5573#issuecomment-1320301162) of adding context where a link opens up a new window.

**Comment by geospatialem** [comment]
Calcite hides icons from assistive technologies so they can't be discerned as they are supplemental to the component, as long as developers follow best practice and provide context in components they will be in alignment with WCAG's success criterion (_refer to [Calcite's accessibility page](https://developers.arcgis.com/calcite-design-system/foundations/accessibility/) for additional developer guidance_).

WRT the above issue, additional context was added to the Link component page on how developers can provide context to links that open in a new window mentioned [here](https://github.com/Esri/calcite-design-system/issues/5573#issuecomment-2018853768). 

Verified on the internal next.sites, where tomorrow's documentation release (Sept 12) will provide the context in the [Link component's Accessibility section](https://developers.arcgis.com/calcite-design-system/components/link/#accessibility).

![image](https://github.com/user-attachments/assets/2b96d318-f70a-4a8a-a58e-88cbca8fc714)
